testsuites/CBLTester/CBL_Functional_tests/APITests/test_vectorSearch.py

```diff
@@ -0,0 +1,423 @@
+import pytest
+import time
+import uuid
+import random
+from CBLClient.Database import Database
+from CBLClient.Replication import Replication
+from libraries.testkit import cluster
+from keywords.ClusterKeywords import ClusterKeywords
+from libraries.testkit.admin import Admin
+from keywords import couchbaseserver
+from keywords.MobileRestClient import MobileRestClient
+from keywords.constants import RBAC_FULL_ADMIN
+from CBLClient.VectorSearch import VectorSearch
+from CBLClient.Collection import Collection
+from CBLClient.Document import Document
+
+bucket = "travel-sample"
+sync_function = "function(doc){channel(doc.channels);}"
+sg_admin_url = None
+sg_db = "db"
+sg_blip_url = None
+sg_url = None
+gteSmallDims = 384  # constant for the number of dims of gteSmall embeddings
+liteserv_platform = None
+
+
+@pytest.fixture
+def vector_search_test_fixture(params_from_base_test_setup, params_from_base_suite_setup):
+    global sg_admin_url
+    global sg_db
+    global sg_blip_url
+    global sg_url
+    global liteserv_platform
+    sg_client = sg_url = sg_admin_url = None
+    cluster_config = params_from_base_test_setup["cluster_config"]
+    sg_admin_url = params_from_base_test_setup["sg_admin_url"]
+    sg_url = params_from_base_test_setup["sg_url"]
+    base_url = params_from_base_test_setup["base_url"]
+    sg_blip_url = params_from_base_test_setup["target_url"]
+    liteserv_platform = params_from_base_test_setup["liteserv_platform"]
+
+    random_suffix = str(uuid.uuid4())[:8]
+    scope = '_default'
+    # Names for the four collections
+    def_col = '_default'
+    st_col_name = 'searchTerms'
+    dbv_col_name = 'docBodyVectors'
+    iv_col_name = 'indexVectors'
+    aw_col_name = 'auxiliaryWords'
+
+    need_sgw_admin_auth = params_from_base_test_setup["need_sgw_admin_auth"]
+    sg_username = "vector_search_user" + random_suffix
+    sg_password = "password"
+    data = {
+        "bucket": bucket, "scopes":
+        {
+            scope: {
+                "collections": {
+                    def_col: {"sync": sync_function},
+                    st_col_name: {"sync": sync_function},
+                    dbv_col_name: {"sync": sync_function},
+                    aw_col_name: {"sync": sync_function},
+                    iv_col_name: {"sync": sync_function}
+                }
+            }
+        }, "num_index_replicas": 0  # This might need to change idk what vectors are
```

Comment:
Does this need changing now?

Reply:
Let's keep it as is for this phase. We might. need to change it later.